packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -19,7 +20,7 @@ import type { InternalTreeNode, Unhydrated } from "./types.js";
 export type TreeNodeSchema<
 	Name extends string = string,
 	Kind extends NodeKind = NodeKind,
-	TNode = unknown,
+	TNode extends TreeNode | TreeLeafValue = TreeNode | TreeLeafValue,
```

Comment:
This is really the main part of this change. Pretty much everything else is done to allow this change to take place.